CodeEdit/Features/Settings/SettingsView.swift

```diff
@@ -7,47 +7,128 @@
+        let lowercasedSearchText: String = searchText.lowercased()
+        var returnedPages: [SettingsPage] = []
+        var foundPage: Bool = false
+
+        for item in pages where item.name == page.name {
+            if item.isSetting && item.displayName.lowercased().contains(lowercasedSearchText) {
+                returnedPages.append(item)
+            } else if item.name.rawValue.contains(lowercasedSearchText) && !item.isSetting {
+                foundPage = true
+            }
+        }
+
+        return SettingsSearchResult(pageFound: foundPage, pages: returnedPages)
+    }
+
+    /// Creates a SettingsPage and it's respective child settings
+    private func createPageAndSettings(
+        _ settingsStruct: Any,
+        parent: SettingsPage,
+        prePages: [SettingsPage]
+    ) -> [SettingsPage] {
+        var pages: [SettingsPage] = prePages
+        pages.append(parent)
+
+        for setting in SettingsData().propertiesOf(settingsStruct) {
+            pages.append(
+                SettingsPage(
+                    parent.name,
+                    baseColor: parent.baseColor,
+                    icon: parent.icon,
+                    isSetting: true,
+                    displayName: setting.nameString
+                )
+            )
+        }
+
+        return pages
+    }
+
+    /// Creates all the neccessary pages
+    private func populatePages() -> [SettingsPage] {
+        var pages: [SettingsPage] = []
+        let settingsData: SettingsData = SettingsData()
+
+        pages = createPageAndSettings(
+            settingsData.general,
+            parent: SettingsPage(.general, baseColor: .gray, icon: .system("gear")),
+            prePages: pages
+        )
+        pages = createPageAndSettings(
+            settingsData.accounts,
+            parent: SettingsPage(.accounts, baseColor: .blue, icon: .system("at")),
+            prePages: pages
+        )
+        pages = createPageAndSettings(
+            settingsData.theme,
+            parent: SettingsPage(.theme, baseColor: .pink, icon: .system("paintbrush.fill")),
+            prePages: pages
+        )
+        pages = createPageAndSettings(
+            settingsData.textEditing,
+            parent: SettingsPage(.textEditing, baseColor: .blue, icon: .system("pencil.line")),
+            prePages: pages
+        )
+        pages = createPageAndSettings(
+            settingsData.terminal,
+            parent: SettingsPage(.terminal, baseColor: .blue, icon: .system("terminal.fill")),
+            prePages: pages
+        )
+        pages = createPageAndSettings(
+            settingsData.sourceControl,
+            parent: SettingsPage(.sourceControl, baseColor: .blue, icon: .symbol("vault")),
+            prePages: pages
+        )
+        pages.append(SettingsPage(.location, baseColor: .green, icon: .system("externaldrive.fill")))
+
+        return pages
+    }
```

Comment:
This is a very confusing pattern to build your pages. Consider letting the `createPageAndSettings` function return an array and combine all arrays afterwards

Reply:
How would that look like in code?